tests/FSharp.Compiler.ComponentTests/Language/DiscriminatedUnionTests.fs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace Language
+
+open FSharp.Test.Compiler
+
+module DiscriminatedUnionTests =
```

Comment:
Can you please add tests for:

* Type that uses `NullAsTrueValue` (the tester should be a static method, not an instance one). I suppose this will work based on the usage of `TDSwitch`, should be fine.

* Types that use `DefaultAugmentation(false)`

* What it does to `ValueOption`, since ValueOption defines it's own tester properties and it does not use DefaultAugmentation